spire-flex/tests/agentHealthCheckReadyPath.yaml

```diff
@@ -20,6 +20,11 @@ tests:
             }
             
             plugins {
+              KeyManager "disk" {
+                plugin_data {
+                  directory = "/opt/spire/data/agent/"
+                }
+              }
```

Comment:
This was the kind of copy/paste changes to every test needed when adding a single feature I was worried about. Its not a lot of text, but its a lot of changes every feature. Really a bad requirement of the unit testing framework...

Reply:
If it gets unbearable, or we have extra time after the deployment is functional, let's revisit this.

I agree it's not ideal, and is a side-effect of wanting the configuration to be in HCL, which requires it to be in a YAML string, which then is just outside of the string tests available at https://github.com/helm-unittest/helm-unittest/blob/main/DOCUMENT.md.

I have done more testing and research around this area than I'd like to admit.  I'll see if I can capture it elsewhere so we don't lose that effort.  One approach is regex (which I dislike) another is to extend helm-unittest, and a third is to attempt build our own testing suite.  As all of these take time, and we lack a deployment, that's why I'm going to bear the cost of the updates till our bigger priorities are settled.

Reply:
I'm wondering if we should support two unit test frameworks for now. We could import the go one from hardened, and use it for the configmap related stuff and use helm-unittest for the rest.

Reply:
As discussed in zoom, we can revisit this later too.

Reply:
(Documenting the zoom meeting's conclustion)

I'm flexible on the point adding in other testing frameworks later, but this submission stands on its own without requiring an additional testing framework; and, there is a high chance that the number of tested configuration files is large, but not too large to complete the task along this path without an additional testing framework.

The current resolution is to actually build up more tests under the current framework, and add in additional testing frameworks in the future if deemed necessary, with the advantage of having most of the tests the framework would need to execute already formed.

Please resolve this discussion.